lib/xlat_tables_v2/xlat_tables_private.h

```diff
@@ -108,6 +108,13 @@ typedef struct {
 	/* Set to 1 when the translation tables are initialized. */
 	int initialized;
 
+	/*
+	 * Bit mask that has to be ORed to the rest of a translation table
+	 * descriptor in order to prohibit execution of code at the exception
+	 * level of this translation context.
+	 */
+	uint64_t execute_never_mask;
```

Comment:
AFAIU, a `xlat_ctx_t` is associated with a unique translation regime. If so, then would it not be more scalable to add a field in this structure that indicates the EL or the translation regime it corresponds to. If the EL is then passed to `xlat_desc()` then the choice between PXN and XN could be made there.
This way if EL3 is creating translation tables on behalf of S-EL1, then there will be no reliance on the `CurrentEL`. Any future EL specific differences between translation regimes could be catered for as well.
Please let me know if I am misunderstanding anything